peer_test.go

```diff
@@ -697,6 +697,118 @@ func TestPeerSelectionRanking(t *testing.T) {
 	}
 }
 
+func TestZeroPeerConnectionCount(t *testing.T) {
+	ch := testutils.NewClient(t, nil)
+	defer ch.Close()
+	err := ch.Peers().SetPeerConnectionCount(0)
+	require.Error(t, err, "peerConnectionCount should not accept 0")
+}
+
+func TestPeerRandomSampling(t *testing.T) {
+	const numIterations = 1000
+
+	testCases := []struct {
+		numPeers            int
+		peerConnectionCount uint32
+		distMin             float64
+		distMax             float64
+	}{
+		// the higher `peerConnectionCount` is, the smoother the impact of uneven scores
+		// become as we are random sampling among `peerConnectionCount` peers
+		{numPeers: 10, peerConnectionCount: 1, distMin: 1000, distMax: 1000},
+		{numPeers: 10, peerConnectionCount: 5, distMin: 160, distMax: 240},
+		{numPeers: 10, peerConnectionCount: 10, distMin: 50, distMax: 150},
+		{numPeers: 10, peerConnectionCount: 15, distMin: 50, distMax: 150},
+	}
+
+	for _, tc := range testCases {
+		// Selected is a map from rank -> [peer, count]
+		// It tracks how often a peer gets selected at a specific rank.
+		selected := make([]map[string]int, tc.numPeers)
```

Comment:
I am not sure if I follow. This is a similar test as `TestPeerSelectionRanking` where we are checking the distribution of the ranking after `numIterations` which means we have to do the checking outside of the loop. Or am I missing something here?